### ISSUE TITLE
Relax some CSS animation constraints

### ISSUE THREAD

**Issue by lannka** [ampproject/amphtml#15194]
For sake of performance, we mostly only allow GPU friendly animations, so only a handful CSS properties are allowed to animate.

We used to rely on an AMP special CSS class named `.amp-animate` to control the start/stop of an animation by the addition/removal of the particular class on any DOM element to animate. However, this mechanism was deprecated and replaced by using CSS `animation-play-state: paused;` (see discussion #7846).

The [CSS selector constraints](https://github.com/ampproject/amphtml/blob/master/extensions/amp-a4a/amp-a4a-format.md#css-animations-and-transitions) were introduced together with this `.amp-animate` class. Any selector containing `.amp-animate` class can only have animation related properties, so that runtime is able to stop animation without other unwanted visual effects.

Given the removal (#11769) of `.amp-animate` class, we don't need this CSS selector constraints any more. i.e., developer should be able to have a CSS selector like:
```
.box {
    color: red;
    transform: rotate(180deg);
    transition: transform 2s;
}
```
/cc @alanorozco @dvoytenko 

TODOs:
- [ ] remove validation rules
- [ ] update documentation

**Comment by honeybadgerdontcare** [comment]
@lannka To be sure I understand, you want all constraints removed or only add `color` to the allowed properties?

The code block gives the following validation error:
```
CSS syntax error in tag 'style amp-custom (AMP4ADS)' - the property 'color' is disallowed together with 'transition'. Allowed properties: ['animation', 'opacity', 'transform', 'transition', 'visibility'].
```

**Comment by lannka** [comment]
We don't need this entire rule that prevents having animation related properties together with other properties under the same CSS selector.

**Comment by honeybadgerdontcare** [comment]
Fixed in #15331 should be live in 1-2 weeks.
